### PR TITLE
added external schema codegen tool documentation

### PR DESCRIPTION
**Pull request by alastairdglennie**
Copied across from [code PR](https://github.com/spatialos/UnrealGDK/pull/879)

### CODE REVIEW

File: SpatialGDK/Documentation/content/helper-scripts.md [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-
+| `BuildWorker.bat` | `<target> <platform> <configuration> <YourGame>.uproject [--skip-codegen]` | Example: </br> `BuildWorker.bat ExampleGameEditor Win64 Development ExampleGame.uproject` </br></br> Build, cook and zip your Unreal server-workers and client-workers for use with a SpatialOS cloud deployment (uploaded using [`spatial cloud upload`](https://docs.improbable.io/reference/latest/shared/deploy/deploy-cloud)). </br></br>  The following `<target>`s  generate zipped workers: </br> - `<YourGame>` </br> - `<YourGame>Server` </br> - `<YourGame>Editor` </br></br> Any other `<target>` passes all arguments to `Engine\Build\BatchFiles\Build.bat` with no cooking or zipping performed.|
```
Comment (ElleEss) [inline SpatialGDK/Documentation/content/helper-scripts.md]: We need the "older" (in fact newer) text for `BuildWorker.bat` which is in this file already.

```suggestion
| `BuildWorker.bat` | `<target> <platform> <configuration> <YourGame>.uproject [--skip-codegen]` | Run this script from the command line to [build your game’s Unreal server-workers and client-workers]({{urlRoot}}/content/glossary#workers) ready for uploading as a SpatialOS [cloud deployment]({{urlRoot}}/content/glossary#deployment) or set them up for testing in a local deployment in the Unreal Editor on your development machine. <br/><br/> Some parameters also [cook](https://docs.unrealengine.com/en-US/Engine/Deployment/Cooking) your workers to Unreal format and zip them up. (A SpatialOS cloud deployment requires built workers in zipped files.) <br/><br/> (In the example and list below, `<YourGame>` is the name of your Unreal project.) </br></br>**Example**</br> `BuildWorker.bat <YourGame>Editor Win64 Development <YourGame>.uproject` </br></br> **Flags** </br> **`<target>`**</br> This is the output file the built workers are stored in. They are [Unreal’s build target](https://docs.unrealengine.com/en-us/Programming/BuildTools/UnrealBuildTool/TargetFiles) files. The target can be: </br> * `<YourGame>Editor`: Set up server-workers for a local deployment for testing in the Unreal Editor on your development machine. This option does not cook or zip server-workers as this is not needed for local deployments. </br> * `<YourGame>Server`: Build, cook and zip server-workers ready for upload to the SpatialOS cloud as a cloud deployment. </br> * `<YourGame>`: Cooks a stand-alone version of the game to test it as a game client. </br></br> If you specify other Unreal  `<target>`, this script passes all arguments to the UE script, `\Engine\Build\BatchFiles\Build.bat`with no cooking or zipping. </br></br> **`<platform>`**</br> Specify the platform your built server-worker or game client to runs on. This is:</br> *  `linux` for your server-workers </br> * `win64` for you game client exectuable </br></br> **`<configuration>`**</br> Use the [Unreal build configuration](https://docs.unrealengine.com/en-us/Programming/Development/BuildConfigurations) `Development` which is the one you usually use during game development. </br></br>* `--skip-codegen` is optional, add this if you want the build script to _NOT_ generate [schema]({{urlRoot}}/content/glossary#schema-generation). |

```

**Comment by alastairdglennie** [comment]
@improbable-davidrodriguez

**Comment by oblm** [comment]
There were some broken links that I've just pushed a fix for (see [UNR-1654](https://improbableio.atlassian.net/browse/UNR-1654)).

**Comment by oblm** [comment]
TOC

I don't think it makes sense to bury this major feature in a `workers and load balancing` on its own.

Thoughts @ElleEss?

File: SpatialGDK/Documentation/content/non-unreal-server-worker-types.md [hunk 2]
```diff
@@ -1,75 +1,85 @@
...
+##### Setting up
 
-There is a basic example in the _Examples_ section below. For more examples of how to construct component updates, command requests, and more, see the SpatialOS documentation on  [serialization in the Worker SDK in C’s API](https://docs.improbable.io/reference/latest/capi/serialization).
+Below is a summary of the steps required to use the external schema code generator. There is a more complete example in the [examples section below](#using-the-code-generator-example).
 
-#### Receive data
+* Run the `ExternalSchemaCodegen.bat` file as described in the [helper scripts documentation]({{urlRoot}}/content/helper-scripts) to generate code from a directory containing external schema files.
+* [Regenerate](https://wiki.unrealengine.com/Generate_Visual_Studio_Project) the Visual Studio project files.
+* Instantiate the `ExternalSchemaInterface` class inside your game module. If you want to ensure that the SpatialOS worker connection registers your callbacks to receive initial network operations, you need to register the callbacks inside your game instance's `OnConnected` event callback.
```
Comment (improbable-davidrodriguez) [inline SpatialGDK/Documentation/content/non-unreal-server-worker-types.md]: As a user, I don't understand what "initial network operations" or why would I want to ensure to receive those.

Comment (oblm) [reply SpatialGDK/Documentation/content/non-unreal-server-worker-types.md]: @alastairdglennie ☝️

Comment (alastairdglennie) [reply SpatialGDK/Documentation/content/non-unreal-server-worker-types.md]:
```suggestion
* Instantiate the `ExternalSchemaInterface` class inside your game module. If you want to ensure that the SpatialOS worker connection registers your callbacks to receive the network operations that are sent as soon as your worker connects, you need to register the callbacks inside your game instance's `OnConnected` event callback.
```

Comment (oblm) [reply SpatialGDK/Documentation/content/non-unreal-server-worker-types.md]: @improbable-davidrodriguez ☝️

File: SpatialGDK/Documentation/content/helper-scripts.md [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
 <br/>
 
 <br/>------------<br/>
-_2019-06-27 Page updated with editorial review_
```
Comment (tenevdev) [inline SpatialGDK/Documentation/content/helper-scripts.md]: Why did you change the style of this line? Also remember to update the date when this is approved.

File: SpatialGDK/Documentation/content/non-unreal-server-worker-types.md [hunk 4]
```diff
@@ -1,75 +1,85 @@
...
+Each callback registration function returns a `CallbackId`. You can deregister your callbacks using the `USpatialDispatcher::RemoveOpCallback` function and passing the `CallbackId` parameter that was returned by the corresponding call to one of the callback registration functions.
 
-There is a basic example in the _Examples_ section below. For more examples of how to deserialize the `Worker_Op` type see the SpatialOS documentation on [serialization in the Worker SDK in C’s API](https://docs.improbable.io/reference/latest/capi/serialization).
+There are basic example in the _Examples_ section below. For more examples of how to serialize and deserialize component updates, command requests, and more, see the SpatialOS documentation on  [serialization in the Worker SDK in C’s API](https://docs.improbable.io/reference/latest/capi/serialization). `Worker_ComponentId` and each network operation type are defined in the [Worker SDK in C’s API](https://docs.improbable.io/reference/latest/capi/api-reference).
```
Comment (tenevdev) [inline SpatialGDK/Documentation/content/non-unreal-server-worker-types.md]: Convert plural to singular or vice versa.

**Comment by ernopp** [comment]
shouldnt this get merged into docs-next?

File: SpatialGDK/Documentation/content/helper-scripts.md (hunk 3 above)
Comment (ernopp) [reply SpatialGDK/Documentation/content/helper-scripts.md]: updated

File: SpatialGDK/Documentation/content/non-unreal-server-worker-types.md (hunk 4 above)
Comment (ernopp) [reply SpatialGDK/Documentation/content/non-unreal-server-worker-types.md]: thanks updated

**Comment by ernopp** [comment]
Applied 2 small fixes based on Tencho's feedback
@oblm I agree this TOC is not great but not in scope of this PR, we are going to do a pass on the TOC before release

Given that Laura has approved merging this in. Thanks Ally for writing this and everyone for reviewing <3
